game/botapi.php

```diff
@@ -210,4 +225,4 @@ function BotResearch ($obj_id)
     else return 0;
 }
 
-?>
```

Comment:
It is better not to insert any characters (line feeds, spaces) after `?>`, as it can play a bad joke.

https://stackoverflow.com/questions/6126554/find-extra-space-new-line-after-a-closing-php-tag

Specifically in this place there is no problem, because the module is purely API, but in the `pages` folder, where the HTML code of pages is generated, this will be unacceptable.